### ISSUE TITLE
Single JSON schema for all endpoint properties?

### ISSUE THREAD

**Issue by Stranger6667** [schemathesis/schemathesis#584]
When I was experimenting with Open API links I found out that to work in a more reliable way it may require a single JSON schema for all endpoint parts - `headers`, `query`, `body`, etc.

For example, we want to run additional tests on endpoint X and take some data from the responses from endpoint Y. To enable shrinking we need to generate these new requests, thus simple replacing values in e.g. `query` will not work. Then we reuse our implementation where `st.builds` generates instances of `Case` from multiple components. These components are independent strategies themselves which leads to a problem:
1. We want to take data from different part of a response / request
2. We want all of them to be present in a generated data
3. It can't be done since `st.builds` arguments are independent

In other words if we have:
- requests query params: `[{"q": 1}, {"q": 2}, {"q": 3}]`
- responses bodies: `[{"user_id": 5}, {"user_id": 6}, {"user_id": 7}]
Then we want that generated requests contain certain combinations that were present in the same request/response:
- Query / body pair is one of `[{"q": 1}, {"user_id": 5}]`, `[{"q": 2}, {"user_id": 6}]`, `[{"q": 3}, {"user_id": 7}]`

The solution may be creating a single JSON schema and generate components as properties of this schema and then build `Case` from them. In this case, we can use `oneOf` to define needed combos

**Comment by Stranger6667** [comment]
Not relevant, since #805 solves it
